internal/controller/utils/sync/nfspvc.go

```diff
@@ -182,3 +184,16 @@ func preparePv(nfspvc danaiov1alpha1.NfsPvc) corev1.PersistentVolume {
 	}
 	return pv
 }
+
+// verifyPvcUid returns true if the pvc uid and the claimRef uid of the pv are not equal.
+func verifyPvcUid(ctx context.Context, k8sClient client.Client, uid types.UID, nfspvc danaiov1alpha1.NfsPvc) bool {
+	pvc := corev1.PersistentVolumeClaim{}
+	if err := k8sClient.Get(ctx, types.NamespacedName{Namespace: nfspvc.Namespace, Name: nfspvc.Name}, &pvc); err != nil {
+		if errors.IsNotFound(err) {
+			return true
+		} else {
+			return false
+		}
+	}
+	return pvc.GetUID() != uid
+}
```

Comment:
The name of the function can be better I think, maybe something like `isUIDEqual`? Also I think it would be less confusing to return `true` if they are equal as opposed to vice-versa